test/blackbox-tests/test-cases/pkg/disjunction.t

```diff
@@ -0,0 +1,51 @@
+Exercise the behaviour of the solver when packages contain disjunctions in
+their dependencies.
+
+  $ . ./helpers.sh
+  $ mkrepo
+
+Create a package that depends on a disjunction of 3 other packages.
+  $ mkpkg a 0.1
+  $ mkpkg b 0.1
+  $ mkpkg c 0.1
+  $ mkpkg dep-on-disjunction 0.1 << EOF
+  > depends: [ "a" | "b" | "c" ]
+  > EOF
+
+The first package in the disjunction is chosen in this case.
+  $ solve dep-on-disjunction
+  Solution for dune.lock:
+  - a.0.1
+  - dep-on-disjunction.0.1
+
+Release a new version of the second package in the disjunction to demonstrate
+that relative version numbers don't affect the choice.
+  $ mkpkg b 0.2
+  $ solve dep-on-disjunction
+  Solution for dune.lock:
+  - a.0.1
+  - dep-on-disjunction.0.1
+
+New version of dep-on-disjunction specifying version numers of deps. Note that
+only "b" exists with the specified version. If the solver chooses the new
```

Comment:
Could you also add `c.0.2` or make it depend on `c.0.1`? Because otherwise `b.0.2` is the only solution and that just tests that the solver can find a solution.

Reply:
I don't think that's true. For example the solver is free to choose `deps-on-disjunction.0.1` rather than `deps-on-disjunction.0.2`.